infer/SigSuggestion.cc

```diff
@@ -89,30 +107,52 @@ void extractSendArgumentKnowledge(core::Context ctx, core::LocOffsets bindLoc, c
         snd->receiverLoc,
         snd->argLocs,
     };
+
+    auto numPosArgs = snd->numPosArgs;
+    auto numKwArgs = snd->args.size() - numPosArgs;
+    bool hasKwSplat = numKwArgs & 0x1;
+
     // Since this is a "fake" dispatch and we are not going to display the errors anyway,
     // core::Loc::none() should be okay here.
     auto originForUninitialized = core::Loc::none();
     auto originForFullType = core::Loc::none();
-    core::DispatchArgs dispatchArgs{snd->fun,       locs,           snd->numPosArgs,
+    core::DispatchArgs dispatchArgs{snd->fun,       locs,           numPosArgs,
                                     args,           snd->recv.type, {snd->recv.type, {originForFullType}},
                                     snd->recv.type, snd->link,      originForUninitialized};
     auto dispatchInfo = snd->recv.type.dispatchCall(ctx, dispatchArgs);
 
-    int i = -1;
-
     // See if we can learn what types should they have
-    for (auto &arg : snd->args) {
-        i++;
+    bool inKwArgs = false;
+    for (int i = 0, argIdx = 0; i < snd->args.size(); i++, argIdx++) {
+        inKwArgs = inKwArgs || (i >= numPosArgs && !hasKwSplat);
```

Comment:
The previous for loop was incrementing `i` for each element of `snd->args`, which would treat the inlined keyword argument pairs the same way as normal positional arguments.